front/temporal/usage_queue/workflows.ts

```diff
@@ -0,0 +1,14 @@
+import { proxyActivities, sleep } from "@temporalio/workflow";
+
+import type * as activities from "@app/temporal/usage_queue/activities";
+
+const { recordUsageActivity } = proxyActivities<typeof activities>({
+  startToCloseTimeout: "10 minutes",
+});
+
+export async function updateWorkspaceUsageWorkflow(workspaceId: string) {
+  // Sleep for one hour before computing usage.
+  await sleep(60 * 60 * 1000);
```

Comment:
why the sleep since we check the rate limit?

Reply:
The sleep function is here to address the issue of potentially having multiple messages following a single message, aiming to capture a more extended period before reporting. An example w/o the sleep function:
- At T - 0, User 1 posts a message, we enqueue an event (user is not yet considered as MAU).
- At T - 1, User 1 posts 5 more messages. The system doesn't enqueue these messages due to the rate limiter, even though the user should have been now considered as MAU.

The sleep function resolved this issue.

Reply:
:+1: not obvious, would move your above comment in code